src/include/stir/recon_buildblock/DataSymmetriesForBins_PET_CartesianGrid.inl

```diff
@@ -369,13 +369,13 @@ find_sym_op_general_bin(
           return new SymmetryOperation_PET_CartesianGrid_swap_zq(view180, axial_pos_shift, z_shift, transform_z);
         else*/  
           if ( do_symmetry_swap_s && s < 0) 
-            return new SymmetryOperation_PET_CartesianGrid_swap_xmx_ymy(view180, axial_pos_shift, z_shift);
+            return new SymmetryOperation_PET_CartesianGrid_swap_xmx_ymy(view180, axial_pos_shift, z_shift, proj_data_info_ptr->get_scanner_ptr()->get_max_num_timing_poss());
           else        
             return new SymmetryOperation_PET_CartesianGrid_swap_zq(view180, axial_pos_shift, z_shift, transform_z);   // s > 0
       }  
       else // segment_num = 0 
       {
-        if ( do_symmetry_swap_s && s < 0) return new SymmetryOperation_PET_CartesianGrid_swap_xmx_ymy(view180, axial_pos_shift, z_shift);
+        if ( do_symmetry_swap_s && s < 0) return new SymmetryOperation_PET_CartesianGrid_swap_xmx_ymy(view180, axial_pos_shift, z_shift, proj_data_info_ptr->get_scanner_ptr()->get_max_num_timing_poss());
```

Comment:
I guess the problem is happening  in one of these modifications

Reply:
Actually not sure anymore because the test is giving errors when the 2 bins are actually the same

Reply:
can you paste some sample errors, for both TOF and non-TOF?

Reply:
WARNING: FactoryRegistry:: overwriting previous value of key in registry.
     key: None
Tests for DataSymmetriesForBins_PET_CartesianGrid
Testing span=1

INFO: Detected view-mashing factor 24 from the number of views (8) and the number of detectors per ring (384).
Adjusting the azimuthal angle offset accordingly (an extra offset of 10.7813 degrees)
	Tests with usual image size

INFO: Determined voxel size by dividing default_bin_size (3.12932) by zoom
		Testing with all symmetries

WARNING: Disabling symmetries as image is rotated due to phi offset of the scanner.
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 0, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 0, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 0, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 0, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 1, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 1, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 1, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 1, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 2, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 2, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 2, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 2, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 3, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 3, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 3, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 3, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 4, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 4, 	view = 0, 	tangential_pos_num = -6, timing position index = 0
Error. Comparing symmetry bin configuration

Current bin: 	segment = -5, 	axial pos 4, 	view = 0, 	tangential_pos_num = -3, timing position index = 0
Symm bin: 		segment = -5, 	axial pos 4, 	view = 0, 	tangential_pos_num = -3, timing position index = 0